packages/shorebird_cli/lib/src/commands/doctor_command.dart

```diff
@@ -57,14 +61,29 @@ Engine • revision ${shorebirdEnv.shorebirdEngineRevision}''',
 
     if (verbose) {
       final notDetected = red.wrap('not detected');
+      var javaVersion = notDetected;
+
+      final javaExe = java.executable;
+      if (javaExe != null) {
+        final javaVersionProcessResult = process.runSync(javaExe, ['-version']);
+        if (javaVersionProcessResult.exitCode == ExitCode.success.code) {
+          // The version string is printed to stderr for some reason.
+          javaVersion = javaVersionProcessResult.stderr
+              .toString()
+              .split(Platform.lineTerminator)
+              .join('${Platform.lineTerminator}                  ');
```

Comment:
It might be good to either derive this number of spaces from the string we're using as a padding with (I believe "JAVA_VERSION") or leave a comment explaining the significance of the spaces.